### ISSUE TITLE
Using folder other than default

### ISSUE THREAD

**Issue by patrickleet** [andrewdavey/cassette#126]
In a project called Web.Ux, not using Bundle.Reference anywhere yet, when I switch Configure to point to "Content" and "Scripts" (after I created them), the application started fine with no error. I tried entering an invalid directory as well, and got a directory not found error -> "Directory not found: D:/xyz/web/Web.Ux/ux/ux/js". Not sure if it will help you in debugging but it's a hybrid webforms/mvc3 application, .net4

```
public void Configure(BundleCollection bundles, CassetteSettings settings)
{
    bundles.AddPerIndividualFile<StylesheetBundle>("ux/js");
    bundles.AddPerIndividualFile<ScriptBundle>("ux/js");
}
```

Folder structure:
Web.UX
- /ux
- \- /js
- \- /css
- \- /pages
- \- - /feature
- \- - - default.aspx
- /Controllers
- /Views

Server Error in '/ux' Application.
Referenced path must be application relative and start with a "~".
Parameter name: path
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.ArgumentException: Referenced path must be application relative and start with a "~".
Parameter name: path

Source Error:

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Stack Trace:

```
[ArgumentException: Referenced path must be application relative and start with a "~".
Parameter name: path]
   Cassette.AssetReference.ValidatePath(String path, AssetReferenceType type) +239
   Cassette.FileAsset.AddBundleReference(String appRelativeFilename, Int32 lineNumber) +106
   Cassette.BundleProcessing.ParseReferences`1.ParseAssetReferences(IAsset asset) +255
   Cassette.BundleProcessing.ParseReferences`1.Process(T bundle, CassetteSettings settings) +119
   Cassette.BundleProcessing.MutablePipeline`1.Process(T bundle, CassetteSettings settings) +520
   Cassette.Configuration.BundleContainerFactoryBase.ProcessAllBundles(IEnumerable`1 bundles, CassetteSettings settings) +275
   Cassette.Configuration.BundleContainerFactory.Create(IEnumerable`1 unprocessedBundles, CassetteSettings settings) +43
   Cassette.CassetteApplicationBase..ctor(IEnumerable`1 bundles, CassetteSettings settings) +64
   Cassette.Web.StartUp.CreateCassetteApplication() +1043
   Cassette.CassetteApplicationContainer.CreateApplication() +97
   System.Lazy`1.CreateValue() +12770415
   System.Lazy`1.LazyInitValue() +355
   Cassette.Web.StartUp.PostApplicationStart() +589

[TargetInvocationException: Exception has been thrown by the target of an invocation.]
   System.RuntimeMethodHandle._InvokeMethodFast(IRuntimeMethodInfo method, Object target, Object[] arguments, SignatureStruct& sig, MethodAttributes methodAttributes, RuntimeType typeOwner) +0
   System.RuntimeMethodHandle.InvokeMethodFast(IRuntimeMethodInfo method, Object target, Object[] arguments, Signature sig, MethodAttributes methodAttributes, RuntimeType typeOwner) +152
   System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture, Boolean skipVisibilityChecks) +396
   System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) +38
   System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters) +35
   WebActivator.BaseActivationMethodAttribute.InvokeMethod() +341
   WebActivator.ActivationManager.RunActivationMethods() +787
   WebActivator.ActivationManager.RunPostStartMethods() +38
   WebActivator.StartMethodCallingModule.Init(HttpApplication context) +159
   System.Web.HttpApplication.RegisterEventSubscriptionsWithIIS(IntPtr appContext, HttpContext context, MethodInfo[] handlers) +575
   System.Web.HttpApplication.InitSpecial(HttpApplicationState state, MethodInfo[] handlers, IntPtr appContext, HttpContext context) +352
   System.Web.HttpApplicationFactory.GetSpecialApplicationInstance(IntPtr appContext, HttpContext context) +407
   System.Web.Hosting.PipelineRuntime.InitializeApplication(IntPtr appContext) +375

[HttpException (0x80004005): Exception has been thrown by the target of an invocation.]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +11686928
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +141
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +4863749
```

**Comment by andrewdavey** [comment]
Thanks for the detailed report. On first glance it looks like a reference in one of your files is causing Cassette to fail. Do you have any `/// <reference path=" "/>` comments in your scripts that look broken?

Are you able to zip up the project (or at least the scripts and stylesheets) and email to me andrew@equin.co.uk ? I can try to determine the problem then.

**Comment by patrickleet** [comment]
I'll take a look through them all, there are probably about 100 js files alone so, I wouldn't be surprised if there is one wrong or outdated in there somewhere. Even though I'm not referencing the files yet it would still cause that error?

**Comment by patrickleet** [comment]
ahh, I commented out the scriptbundle line and it is loading now, so it must be a reference somewhere. I didn't realize that referencing the folder would parse all of the files, I figured it wouldn't do so until each file is referenced individually

**Comment by patrickleet** [comment]
can you have reference paths such as this: 

/// &lt;reference path="../libraries/jquery.js" /&gt;

Folder structure:
/js
- /feature
- \- file.js
- /libraries
- \- jquery.js

**Comment by andrewdavey** [comment]
Yeah, I am confused about the error you're getting. It's not what I'd expect for a simple incorrect file reference. If you run with the VS debugger attached, are you able to see the value of the `path` parameter? That may help track down the origin of the problem.

**Comment by andrewdavey** [comment]
Yep, all bundles are processed up-front during application start up.

Relative asset references should work fine too. The path is relative to the current file.

**Comment by patrickleet** [comment]
I found two files in a subfolder (which I'm not sure are even used anymore) with an old reference to vs-doc and old version of jquery that are no longer existent, once I removed them and added the correct jquery reference I decided that I was pretty sure the rest of the files were correct and tried running again. Sure enough, no errors. So everything seems to be working for now. Thanks for your help.

It's a pretty large project with a lot of old code, so if I run into another problem, I'll post it on the disqus comments in the documentation. Thanks again!
